documentation/docs/announcing-1.0.md

```diff
@@ -0,0 +1,216 @@
+
+Also the ergonomics of the language with variants, pattern matching and option
+types. It's like having ES2040, but with the bad parts removed.
+
+When building a real app an important building block is data fetching. You need
+it for almost any kind of application. There are several ways to go about
+fetching data, but there is one protocol that's becoming more and more
+ubiquitous, and it looks like it has become the successor of the REST protocol.
+
+GraphQL is a great pairing with Reason, because the protocol is **typed**. This
+means that you have a predefined schema of data that your application can fetch.
+In principle when we compose a query, given the schema, we know exactly what the
+types of the data that is returned is going to be.
+
+In my experience the biggest source of bugs is that the data you get back is not
+what you expect. In TypeScript or JavaScript applications you might forget that
+a value can be `null` or you might even forgot to include a certain data
+requirements in your fragment. Best practice is to
+[generate types for your queries](https://www.apollographql.com/docs/react/development-testing/static-typing/).
+But now you have to make sure the types always exactly reflect your query, it's
+just trading one type of bug for another type of bug, while writing a lot more
+code. There are also tools to
+[automatically generate](https://github.com/apollographql/apollo-tooling) types
+from the queries, mutations and fragments in the source code. But now you need
+to import these types from the generated locations, and there is a lot of
+annotation you need to do (also nothing prevents you from a wrong annotation).
+Every time the schema or your query changes you need to remember to rerun the
+generally slow tooling to keep them in sync. The worst part is that you still
+need to
+[create types](https://hasura.io/learn/graphql/typescript-react-apollo/queries/4-mapping-types/),
+for parts of the query that you are manipulating to help the typechecker. My
+experience with it has been that the amount of type annotation and work you have
+to do to keep all the tooling working in many cases makes you so unproductive
+that the better alternative is to live with the unsafety. But that doesn't solve
+the largest source of bugs in your apps.
+
+`graphql-ppx` is trying to make working with graphql definitions seamless, as if
+it's a new **language primitive**. When you define a graphql query, mutation or
+fragment in Reason, `graphql-ppx` will generate all the types. It never needs to
+sync up, and is superfast.
+
+You can write a query like this:
+
+```reason
+[%graphql {|
+  query UserQuery {
+    user {
+      id
+      name
+      type
+    }
+  }
+|}];
+```
+
+This generates the following module (simplified):
+
+```reason
+module type UserQuery = {
+  module Raw: {
+    type t_user_type = string;
+    type t_user = {
+      id: string,
+      name: Js.Nullable.t(string),
+      type: t_user_type
```

Comment:
Does this automatically become `type_` now?

Reply:
Yes but still a typo 🙈